src/library/trackloader.cpp

```diff
@@ -0,0 +1,82 @@
+#include "library/trackloader.h"
+
+#include <QMetaMethod>
+#include <QThread>
+
+#include <mutex>
+
+#include "library/trackcollectionmanager.h"
+#include "util/logger.h"
+
+
+namespace mixxx {
+
+namespace {
+
+const Logger kLogger("TrackLoader");
+
+std::once_flag registerMetaTypesOnceFlag;
```

Comment:
I did not know about this language feature. That's handy :+1:

Reply:
Would be overkill in constructors of basic types like `TrackId` and doesn't work for typedefs like `TrackPointer`. But helps to improve the locality of those boilerplate glue code that easily scattered around out of the original context. Yet still a crutch to overcome C++/Qt deficiencies.